src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Asset,
+        [Parameter(Mandatory=$false)]
+        [object[]]$Tag
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+   
+    # we need to use function Resolve-SafeguardAssetId from the asset.psm1 module to get the Asset ID so check if the module is load
+    if (-not (Get-Module assets)) {
+        Import-module "$PSScriptRoot\assets.psm1"
+    }
+    $AssetId = (Resolve-SafeguardAssetId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId -Asset $Asset)    
+    $local:body =@() # empty array to store the tag IDs.
+    if ($Tag.Count -gt 0) {
+        foreach ($tagObj in $Tag) {
+            $tagId = Resolve-SafeguardTagId $tagObj
+            if ($tagId) {
+                # found the tag so add the ID to the body
+                $local:body += [PSCustomObject]@{ Id = $tagId }
+            }
+        }
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core PUT "Assets/$AssetId/Tags" -Body $local:body
+    } else {
+        # No tags specified so remove all tags. Send empty json body.
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core PUT "Assets/$AssetId/Tags" -JsonBody "[]"
+    }
+}
+
+
+<#
+.SYNOPSIS
+Get the tags from a specific account via the Web API.
+
+.DESCRIPTION
+Get the assigned tags for a specific account.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition to get tags from an asset.
+If this value is empty, the Macrocosm partition will be used.
+
+.PARAMETER AssetPartitionId
+An integer containing the asset partition ID to get the asset's tags from.
+(If specified, this will override the AssetPartition parameter)
+
+.PARAMETER Account
+Mandatory parameter. An integer containing the ID of the account or a string containing the name of the account to get the tags for.
+
+.PARAMETER Field
+String array with the tag property names to return.
+(Can be one of the following: Id,Name,Description,AdminAssigned
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Get-SafeguardAccountTag "accountname" 
+
+.EXAMPLE
+Get-SafeguardAccountTag "accountname" -Field Id,Name,Description
+
+#>
+function Get-SafeguardAccountTag {
```

Comment:
Should we rename this method to `Get-SafeguardAssetAccountTag`? I think that might be more consistent with what is used else where. For example, in `assets.psm1`:
![image](https://github.com/user-attachments/assets/47b1e3da-3ea0-47d8-9011-8b482bf80b37)